sigar_linux.go

```diff
@@ -793,6 +787,25 @@ func readFile(file string, handler func(string) bool) error {
 	return nil
 }
 
+// Read the first line of a file, ignoring any error
+func readFileLine(file string) string {
+	f, err := os.Open(file)
+	if err != nil {
+		return ""
+	}
+	defer f.Close()
+
+	reader := bufio.NewReader(f)
+	line, _ := reader.ReadString('\n')
+	return strings.TrimSpace(line)
+}
+
+// Convert the value to an uint64, ignoring any error
+func readUint(val string) uint64 {
+	result, _ := strconv.ParseUint(val, 10, 64)
+	return result
```

Comment:
This is a great catch - its pretty conventional that any returned values when an error is present are undefined, and so we should not rely on the result being zero or any other value, even though `ParseUint` does define a slightly stronger contract. This gives me the itch to fix all those `strtoull` calls.